client/commonFramework/add-loop-protect.js

```diff
@@ -4,13 +4,16 @@ window.common = (function(global) {
     common = { init: [] }
   } = global;
 
-  loopProtect.hit = function hit(line) {
-     var err = 'Error: Exiting potential infinite loop at line ' +
-     line +
-     '. To disable loop protection, write: \n\\/\\/ noprotect\nas the first' +
-     'line. Beware that if you do have an infinite loop in your code' +
-     'this will crash your browser.';
-    console.error(err);
+  loopProtect.hit = function(line) {
+    var msg = 'Potential infinite loop at line ' +
+      line +
+      '. To disable loop protection, write:' +
+      ' \\n\\/\\/ noprotect\\nas the first' +
+      ' line. Beware that if you do have an infinite loop in your code' +
+      ' this will crash your browser.';
+    window.__err = new Error(msg);
```

Comment:
Throwing the error ensures test failure. But depending on which test fails, you may not see the error. For example, if in the test you put while(true) for the loop, then the error will be shown in the black results box, because we only show one result there. (Which for the Drop It page is the third test.)